packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java

```diff
@@ -0,0 +1,413 @@
+package jmxquery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.Collection;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import javax.management.MBeanInfo;
+import javax.management.MBeanServerConnection;
+import javax.management.MBeanAttributeInfo;
+import javax.management.ObjectName;
+import javax.management.ObjectInstance;
+import javax.management.AttributeList;
+import javax.management.Attribute;
+import javax.management.openmbean.CompositeDataSupport;
+import javax.management.openmbean.CompositeType;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+
+/*                                                                                                                                                                            
+ *  Copyright 2012 Rackspace
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
+/**
+ *
+ * JMXQuery is used for local or remote request of JMX attributes
+ * It requires JRE 1.5 to be used for compilation and execution.
+ * Look method main for description how it can be invoked.
+ *
+ * This plugin was found on nagiosexchange.  It lacked a username/password/role system.
+ *
+ * @author unknown
+ * @author Ryan Gravener (<a href="http://ryangravener.com/app/contact">rgravener</a>)
+ *
+ */
+public class JMXQuery {
+
+  private String url;
+  private int verbatim;
+  private JMXConnector connector;
+  private MBeanServerConnection connection;
+  private String attribute;
+  private ArrayList<String> attributeKeys = new ArrayList<String>();;
+  private String object;
+  private String username, password;
+
+  private static int maxMatchingObjects = 15;
+
+  private Object checkData;
+
+  private void connect() throws IOException
+  {
```

Comment:
Heh, I actually imported this from ck-agent repo unmodified.

It's true that I wrote it, but the changes I did were mostly just modifications and additions to an existing code from jmxquery repo and I just followed an existing style back then :P